files/class-vip-filesystem-stream-wrapper.php

```diff
@@ -694,9 +695,25 @@ public function mkdir( $path, $mode, $options ) {
 	 * @return  bool
 	 */
 	public function stream_metadata( $path, $option, $value ) {
-		$this->debug( sprintf( 'stream_metadata =>  %s + %s + %s', $path, $option, $value ) );
+		$this->debug( sprintf( 'stream_metadata =>  %s + %s + %s', $path, $option, json_encode( $value ) ) );   // phpcs:ignore WordPress.WP.AlternativeFunctions.json_encode_json_encode
 
-		return false;
+		switch ( $option ) {
+			case STREAM_META_TOUCH:
+				if ( false === file_exists( $path ) ) {
+					$file = fopen( $path, 'w' );    // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_fopen
+					if ( is_resource( $file ) ) {
+						fclose( $file );
```

Comment:
What if `fclose()` returns `false`?

Reply:
Then I will be very sad :-(

But jokes aside, the file is empty - there are no writes, I don't think it is possible for the filesystem to return an `ENOSPC` error.

Anyway, `fopen()` and `fclose()` will be handled by the methods of our FS wrapper; in case of an error, they will show error messages; I don't think we need to duplicate them here.

The `is_resource()` check is only to avoid calling `fclose()` on an invalid file descriptor - to avoid further errors.